research.rst

```diff
@@ -0,0 +1,269 @@
+        staging/
+            secrets.yml
+            non-secrets.yml
+        dev/
+            secrets.yml
+            non-secrets.yml
+
+
+and then the ``non-secrets.yml`` files make use of variables defined
+in ``secrets.yml``:
+
+.. code-block:: yaml
+
+    postgres_host: localhost
+    postgres_database: pg_database
+    postgres_user: pg_user
+    postgres_password: {{ VAULT_POSTGRES_PASSWORD }}
+    postgres_port: 5432
+
+
+Also, in the playbook and role tasks, make sure to set ``no_log:
+true`` so that the secrets don't get echoed to the console when the
+verbosity is turned up.
+
+.. code-block:: yaml
+
+    - name: VPN Server | Load VPN secret keys
+      include_vars: "vpn-secrets.yml"
+      no_log: true
+
+
+For ease of use, we could do away with the secret/non-secret file
+split for the dev environment.
+
+Other possible options for sharing the Vault key:
+
+- keep on the intranet (either vault files or keys for files)
+- HashiCorp's Vault
+- consul
+- KeePassX
+
+Sources:
+
+- http://www.slideshare.net/excellaco/using-ansible-vault-to-protect-your-secrets
+
+
+Configuration and Customization
+-------------------------------
+
+Ansible will look relative to the playbook directory or the inventory
+directory for variable files and other such files, as well as in the
+appropriate directories inside roles.  So in order to configure a
+project, it is sufficient to have a set of directories named according
+to convention that will contain needed configuration variables.  A
+likely possibility is
+
+::
+
+    django-project-template/
+        inventory/
+            group_vars/
+            host_vars/
+
+
+This has mostly been done already in the ``use_tequila`` branch,
+though some adjustments should be made in order to follow the
+recommended secrets-vs-non-secrets structure.
+
+The ``ansible.cfg`` that ships with the project will need to define
+the inventory location.
+
+Since the relevant playbook(s) for a project will ship inside that
+project, customized tasks can be added directly in that file.  If
+there are sufficient numbers of these tasks for it to be desirable,
+additional playbooks can be constructed and put in a conventional
+location in the project (e.g. ``playbooks/``), and then brought into
+the main playbook using the ``include`` directive.
+
+
+Dynamic Inventory Management
+----------------------------
+
+As with the password file, Ansible will accept a script for its
+inventory file.  This opens up the possibility of having a *dynamic*
+inventory.  Ansible itself ships with a few working examples,
+including scripts for AWS EC2 and OpenStack.
+
+.. FIXME: Needs more here.
+
+
+Conversion From Margarita
+-------------------------
+
+Needed:
+
+- one-shot playbook to remove Salt from the servers
+- create the directory structure used by the tequila-specific portions
+  of django-project-template
+- skeletons of project-specific Ansible variables files
+- convert existing knowledge about servers into inventory files?
```

Comment:
This could be possible by asking the master for the grain information (current roles, IP, address, etc) for each existing minion. i.e. `$ fab staging salt:"grains.items"` and parsing that information to produce the inventory.